### PR TITLE
Adding new signatures to editor constructors.

### LINKED ISSUE #19887: Passing DOM Element in the config object instead of editor constructor parameter
To implement decisions from the [RFC](https://github.com/ckeditor/ckeditor5/issues/19864) we need to:
* Implement [aggregated editor config](https://github.com/ckeditor/ckeditor5/issues/19885)
* Extend the config normalizer implemented in the config aggregation task with the `element` field.
  * It should accept a DOM element.
  * It should use the editor constructor `sourceElementOrData` parameter as a fallback for the element reference (only if it is a DOM element, or record with DOM elements, not string - not initial data)
* The editor constructors (all types) should get an additional signature without the first parameter. 
  *  The old signatures (with `sourceElementOrData` parameter) should be deprecated.
  * The static `create()` methods should also be updated in the same way as constructors.
* The editor watchdog and context watchdog should be updated
  * An additional method signature without element reference should be added; the old ones should be deprecated.
  * `ContextWatchdog` expects `ContextWatchdogItemConfiguration` with creator function (similar to the editor watchdog) and `sourceElementOrData` property that reflects the first parameter of editor constructor - this field should be marked as deprecated.
* The classic editor should get an additional config option - `attachTo`. It should accept only a DOM element. It should work the same as currently, so it could be used as an initial data provider, not only an element used for replacement with an editor.
  * When not provided, normalizer should fallback to the constructor parameter (if a DOM element is provided).
  * Classic editor should warn if a DOM element is provided in the `root.element` field, as this is not possible in this scenario.
* The multi-root `createEditable()` should also accept the `element` key (DOM element).

### PR DESCRIPTION
**Pull request by niegowski**
### 🚀 Summary

Implements decisions from the https://github.com/ckeditor/ckeditor5/issues/19864

TODO - details will be added later

---

### 📌 Related issues

* Closes #19887

---

### 💡 Additional information

*Optional: Notes on decisions, edge cases, or anything helpful for reviewers.*

---

### 🧾 Checklists

Use the following checklists to ensure important areas were not overlooked.
This does not apply to feature-branch merges.
If an item is **not relevant** to this type of change, simply leave it unchecked.

#### Author checklist

- [ ] Is the changelog entry intentionally omitted?
- [ ] Is the change backward-compatible?
- [ ] Have you considered the impact on different editor setups and core interactions? _(e.g., classic/inline/multi-root/many editors, typing, selection, paste, tables, lists, images, collaboration, pagination)_
- [ ] Has the change been manually verified in the relevant setups?
- [ ] Does this change affect any of the above?
- [ ] Is performance impacted?
- [ ] Is accessibility affected?
- [ ] Have tests been added that fail without this change (against regression)?
- [ ] Have the API documentation, guides, feature digest, and related feature sections been updated where needed?
- [ ] Have metadata files (ckeditor5-metadata.json) been updated if needed?
- [ ] Are there any changes the team should be informed about (e.g. architectural, difficult to revert in future versions or having impact on other features)?
- [ ] Were these changes documented (in Logbook)?

#### Reviewer checklist

- [ ] PR description explains the changes and the chosen approach (especially when  performance, API, or UX is affected).
- [ ] The changelog entry is clear, user‑ or integrator-facing, and it describes any breaking changes.
- [ ] All new external dependencies have been approved and mentioned in LICENSE.md (if any).
- [ ] All human-readable, translateable strings in this PR been introduced using `t()` (if any).
- [ ] I manually verified the change (e.g., in manual tests or documentation).
- [ ] The target branch is correct.

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it changes editor construction/initialization paths across all editor types and watchdog restart logic, which can affect integrations and editor bootstrapping behavior. Backward compatibility is maintained via deprecated overloads but signature detection and root normalization may have edge cases.
> 
> **Overview**
> Adds *config-only* constructor/`create()` overloads across `ClassicEditor`, `InlineEditor`, `BalloonEditor`, `DecoupledEditor`, and `MultiRootEditor`, deprecating the legacy “first argument is element/data” signatures while keeping them supported.
> 
> Extends root normalization to support `RootConfig.element` and `ClassicEditor`’s new `config.attachTo` source element, including warnings for unsupported/ignored combinations (e.g. `attachTo` on non-classic editors, `root.element` conflicting with `attachTo`, and ignoring `addRoot({ element })` in multi-root).
> 
> Updates watchdogs to allow editor creation *without* `sourceElementOrData` (config-based creator mode), including restart behavior and signature detection, and adds/adjusts tests (plus a drag&drop finalization edge-case test and removal of a redundant integration suite).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 09dafa3447a342f857f2384431e978e026bc672c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/ckeditor5-core/tests/_utils/classictesteditor.js [hunk 1]
```diff
@@ -20,31 +20,24 @@ export class ClassicTestEditor extends ElementApiMixin( Editor ) {
...
-			this.model.schema.extend( '$text', {
-				allowIn: '$inlineRoot'
-			} );
+		if ( isElement( sourceElement ) ) {
+			this.sourceElement = sourceElement;
 		}
 
 		// Create the ("main") root element of the model tree.
-		this.model.document.createRoot( isInline ? '$inlineRoot' : '$root' );
+		this.model.document.createRoot();
```
[the remark this answers is not in the document]
Comment (niegowski) [reply packages/ckeditor5-core/tests/_utils/classictesteditor.js:40]: That was an invalid test. I need to add a proper test in the clipboard package.
